OOP-functions/oop/app.js

```diff
@@ -0,0 +1,426 @@
+class App extends Component {
+    constructor() {
+        super();
+        this.state = {
```

Comment:
please read about immutable data and how to use useState in react, it's gonna be interesting for you